static/js/src/navigation.js

```diff
@@ -1,91 +1,209 @@
-var navDropdowns = [].slice.call(
-  document.querySelectorAll(".p-navigation__dropdown-link")
+const navDropdowns = [].slice.call(
+  document.querySelectorAll(
+    ".p-navigation__item--dropdown-toggle:not(.global-nav__dropdown-toggle)"
+  )
 );
-var dropdownWindow = document.querySelector(".dropdown-window");
-var dropdownWindowOverlay = document.querySelector(".dropdown-window-overlay");
+const dropdownWindow = document.querySelector(".dropdown-window");
+const dropdownWindowOverlay = document.querySelector(
+  ".dropdown-window-overlay"
+);
+const secondaryNav = document.querySelector(".p-navigation.is-secondary");
+const navigation = document.querySelector(".p-navigation");
 
 navDropdowns.forEach(function (dropdown) {
   dropdown.addEventListener("click", function (event) {
     event.preventDefault();
 
     var clickedDropdown = this;
 
-    dropdownWindow.classList.remove("slide-animation");
-    dropdownWindowOverlay.classList.remove("fade-animation");
+    handleDropdownClick(clickedDropdown);
+  });
+});
 
-    navDropdowns.forEach(function (dropdown) {
-      var dropdownContent = document.getElementById(dropdown.id + "-content");
+document.addEventListener("global-nav-opened", () => {
+  dropdownWindow.classList.add("slide-animation");
+  dropdownWindowOverlay.classList.add("fade-animation");
 
-      if (dropdown === clickedDropdown) {
-        if (dropdown.classList.contains("is-selected")) {
-          closeMenu(dropdown, dropdownContent);
-        } else {
-          dropdown.classList.add("is-selected");
-          dropdownContent.classList.remove("u-hide");
+  navDropdowns.forEach((dropdown) => {
+    const dropdownContent = document.getElementById(dropdown.id + "-content");
+    const dropdownContentMobile = document.getElementById(
+      dropdown.id + "-content-mobile"
+    );
 
-          if (window.history.pushState) {
-            window.history.pushState(null, null, "#" + dropdown.id);
-          }
-        }
-      } else {
-        dropdown.classList.remove("is-selected");
-        dropdownContent.classList.add("u-hide");
-      }
-    });
+    dropdown.classList.remove("is-active");
+    dropdownContent.classList.add("u-hide");
+    dropdownContentMobile.classList.add("u-hide");
   });
 });
 
-// Close the menu if browser back button is clicked
-window.addEventListener("hashchange", function () {
-  navDropdowns.forEach(function (dropdown) {
-    const dropdownContent = document.getElementById(dropdown.id + "-content");
+function handleDropdownClick(clickedDropdown) {
+  const isActive = clickedDropdown.classList.contains("is-active");
+
+  dropdownWindow.classList.remove("slide-animation");
+  dropdownWindowOverlay.classList.remove("fade-animation");
+  toggleNavMenu(clickedDropdown, !isActive);
+}
+
+function toggleNavMenu(dropdown, show) {
+  const dropdownContent = document.getElementById(dropdown.id + "-content");
+  const dropdownContentMobile = document.getElementById(
```

Comment:
Might be worth moving to the top? As this seems to be used in several places (line 29)